src/components/courses/editCourseView.tsx

```diff
@@ -2,59 +2,62 @@
 'use client';
 
 import { AdminPageTitle } from '@components/layouts';
-import { transformRawProgram } from '@hooks/useProgram';
 import { Button, Spinner } from 'flowbite-react';
 import i18next from 'i18next';
-import { useSearchParams } from 'next/navigation';
-import { FormEventHandler, useEffect } from 'react';
+import { FormEventHandler, useState } from 'react';
 import { COURSE_FORM_FIELDS } from './constants';
-import { transformRawSchool } from '@hooks/useSchool';
-import useCourse, { transformRawCourse } from '@hooks/useCourse';
+import useCourse from '@hooks/useCourse';
 import { AdminFormInput } from '@components/admin/AdminForm/AdminFormInput';
-import { AdminFormSelector } from '@components/admin/AdminForm/AdminFormSelector';
 import { extractValuesFromFormEvent } from 'utils/helpers';
 import { T_CourseFields } from '@hooks/useCourse/types';
-import { API_LINKS } from 'app/links';
-import { useAdminListViewData } from '@hooks/useAdmin/useAdminListViewData';
-import { useAdminItemById } from '@hooks/useAdmin/useAdminItemById';
 import { LibraryErrorMessage } from '@components/library/LibraryErrorMessage/LibraryErrorMessage';
-
-const EditCourseView = ({ courseId, onSuccessfullyDone }: { courseId?: string; onSuccessfullyDone?: () => void }) => {
-  const { editCourse } = useCourse();
-
-  const searchParams = useSearchParams();
-
-  const { item: course, isLoading } = useAdminItemById(
-    API_LINKS.FETCH_COURSE_BY_ID,
-    courseId || (searchParams.get('courseId') as string),
-    'course',
-    transformRawCourse,
-  );
-
-  const { items: schools, isLoading: loadingSchools } = useAdminListViewData(
-    API_LINKS.FETCH_SCHOOLS,
-    'schools',
-    transformRawSchool,
-  );
-
-  const { items: programs, isLoading: loadingPrograms } = useAdminListViewData(
-    API_LINKS.FETCH_PROGRAMS,
-    'programs',
-    transformRawProgram,
-  );
-
-  const handleSubmit: FormEventHandler<HTMLFormElement> = (e) => {
-    e.preventDefault();
-
-    const keys = [
-      COURSE_FORM_FIELDS.name.key,
-      COURSE_FORM_FIELDS.description.key,
-      COURSE_FORM_FIELDS.school.key,
-      COURSE_FORM_FIELDS.program.key,
-    ];
-
-    let result = extractValuesFromFormEvent<T_CourseFields>(e, keys);
-    editCourse({ ...course, ...result }, onSuccessfullyDone);
+import { DeletionWarningModal } from '@components/admin/AdminTable/DeletionWarningModal';
+
+const EditCourseView = ({ course, onSuccessfullyDone }: { course: T_CourseFields; onSuccessfullyDone: () => void }) => {
+  const { editCourse, deleteCourse } = useCourse();
+  const [uploading, setUploading] = useState(false);
+  const [showDeletionWarning, setShowDeletionWarning] = useState(false);
+  const toggleDeletionWarning = () => setShowDeletionWarning((value) => !value);
+
+  // const searchParams = useSearchParams();
+
+  // const { item: course, isLoading } = useAdminItemById(
```

Comment:
Since there's the other [issue](https://github.com/SparkEdUAB/sparked-next/issues/73#issuecomment-2116621093) of having to fetch extra details of a specific item, it means we'll have to fetch the item's data individually again instead of passing it as a prop as I've done in this change. So we'll have to restore it partially, though with some changes